### PR TITLE
fix: collapse default position in rtl

### LINKED ISSUE #23444: Collapse icon in rtl
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://ant.design/components/collapse-cn/](https://ant.design/components/collapse-cn/)

### Steps to reproduce
- LTR 模式下，若未设置 expandIconPosition ，默认显示在左边
- 切换 RTL 后， icon  仍在左边

### What is expected?
- 显示右边

### What is actually happening?
- 显示左边

| Environment | Info |
|---|---|
| antd | 4.1.4 |
| React | The latest |
| System | mac |
| Browser | chrome |

### PR DESCRIPTION
**Pull request by xrkffgg**
### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
close https://github.com/ant-design/ant-design/issues/23444

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Fix Collapse default icon position in RTL          |
| 🇨🇳 Chinese | 修复 Collapse 在 RTL 下，默认按钮位置        |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️ 

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/collapse/Collapse.tsx [hunk 1]
```diff
@@ -66,10 +65,16 @@ export default class Collapse extends React.Component<CollapseProps, any> {
       expandIconPosition,
     } = this.props;
     const prefixCls = getPrefixCls('collapse', customizePrefixCls);
+    let iconPosition;
+    if (expandIconPosition !== undefined) {
+      iconPosition = expandIconPosition;
+    } else {
+      iconPosition = direction === 'rtl' ? 'right' : 'left';
+    }
```
Comment (afc163) [inline components/collapse/Collapse.tsx:73]: 提到一个函数里。

File: components/collapse/Collapse.tsx [hunk 2]
```diff
@@ -42,6 +42,13 @@ export default class Collapse extends React.Component<CollapseProps, any> {
     bordered: true,
   };
 
+  getIconPosition(isRTL: boolean, expandIconPosition: ExpandIconPosition) {
+    if (expandIconPosition !== undefined) {
+      return expandIconPosition;
+    }
+    return isRTL ? 'right' : 'left';
+  }
```
Comment (afc163) [inline components/collapse/Collapse.tsx:50]: `isRTL` 和 `expandIconPosition` 从 this.props 中计算，不要作为参数传递。

File: components/collapse/Collapse.tsx [hunk 3]
```diff
@@ -40,7 +40,15 @@ export default class Collapse extends React.Component<CollapseProps, any> {
 
   static defaultProps = {
     bordered: true,
-    expandIconPosition: 'left' as CollapseProps['expandIconPosition'],
+  };
+
+  getIconPosition = () => {
+    const { expandIconPosition } = this.props;
+    const { direction } = React.useContext(ConfigContext);
```
Comment (afc163) [inline components/collapse/Collapse.tsx]: direction 应该只能用参数传进来。

Comment (xrkffgg) [reply components/collapse/Collapse.tsx]: 嗯，已经改啦~

**Pull request merged**